File: solution_zzj/153._Find_Minimum_in_Rotated_Sorted_Array.cc
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0, r = nums.size()-1;
        while (l <= r) {
            int mid = (l+r)/2;
            if (nums[mid] > nums[nums.size()-1]) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return nums[l];
    }
};

File: Weijia/153. Find Minimum in Rotated Sorted Array.py
class Solution(object):
    def findMin(self, nums):
        s1=0
        s2=len(nums)
        while s1<=s2:
            mid=int((s1+s2)/2)
            if nums[mid-1]<nums[mid]:
                if nums[mid]>nums[-1]:
                    s1=mid+1
                else:
                    s2=mid-1
            else:
                return nums[mid]
        return nums[s1]
